docs/release_process.md

```diff
@@ -0,0 +1,47 @@
+# Release procedure
+
+This document describes how to release a new bpftrace version.
+
+The "release manager" (RM) can be one or more person. Usually whoever is motivated
+enough to drive a release.
+
+## Branching model
+
+In the usual case, we release directly from master. Reasoning is that bpftrace
+isn't a huge project yet so complicated branching models and release strategies
+more get in the way than provide order. If master is really busy or really buggy,
+the RM can choose to cut a release branch (titled `X.Y.Z_release`) to try and
+stabilize the code without including work in progress into the release.
+
+## Merging pull requests
+
+Please squash + rebase all pull requests (with no merge commit). In other words,
+there should be one commit in master per pull request. This makes generating
+changelogs both trivial and precise with the least amount of noise.
```

Comment:
+1 for doing this on the smaller merge requests, but you loose quite some context on the bigger ones if you do this. 

When the author has a cleanly written history AND the change is sufficiently complex we should allow rebase merges (without squash). But I've not found a good way to define sufficiently yet...